testsuite/smoke-test/src/key_manager.rs

```diff
@@ -3,68 +3,66 @@
 
 use crate::{
     smoke_test_environment::SmokeTestEnvironment,
-    test_utils::libra_swarm_utils::{get_json_rpc_libra_interface, load_node_config},
-    workspace_builder,
+    test_utils::libra_swarm_utils::load_backend_storage,
 };
-use libra_config::config::{Identity, KeyManagerConfig};
+use libra_config::config::NodeConfig;
 use libra_global_constants::CONSENSUS_KEY;
-use libra_key_manager::libra_interface::LibraInterface;
+use libra_key_manager::{
+    libra_interface::{JsonRpcLibraInterface, LibraInterface},
+    KeyManager,
+};
 use libra_secure_storage::{CryptoStorage, Storage};
-use std::{convert::TryInto, process::Command, thread::sleep, time::Duration};
-
-const KEY_MANAGER_BIN: &str = "libra-key-manager";
+use libra_secure_time::RealTimeService;
+use libra_smoke_test_attribute::smoke_test;
+use libra_types::chain_id::ChainId;
+use rusty_fork::rusty_fork_test;
+use std::{convert::TryInto, thread, thread::sleep, time::Duration};
 
-#[test]
-#[ignore]
+#[smoke_test]
 fn test_key_manager_consensus_rotation() {
+    // Create and launch a local validator swarm
     let mut env = SmokeTestEnvironment::new(1);
     env.validator_swarm.launch();
```

Comment:
so these are all still running using the legacy method, right?

Reply:
Yes, nothing has changed from the perspective of the swarm. These changes will come in a follow up PR.